Assets/Scripts/TaskGenerator.cs

```diff
@@ -0,0 +1,156 @@
+                    break;
+                case "SolarPanel":
+                    perfabs = GamePrefabs.Where(x => x.name.Contains("Rewire") || x.name.Contains("ColorBeep")).ToList();
+                    gamePrefab = perfabs[random.Next(perfabs.Count)];
+                    break;
+                default:
+                    gamePrefab = GamePrefabs[random.Next(GamePrefabs.Length)];
+                    break;
+
+            }
+
+            AddTaskToObject(gameObject, gamePrefab);
+        }
+    }
+
+    private void AddTaskToObject(GameObject interactableObject, GameObject gamePrefab)
+    {
+        //Adds the script component
+        InteractableObject component = interactableObject.AddComponent<InteractableObject>(); 
+        component.GamePrefab = gamePrefab;
+
+        //Changes the color
+        interactableObject.GetComponent<MeshRenderer>().material = canSolveMaterial;
+    }
+
+    private void ChooseAllTasksAtStart()
+    {
+        //Chooses at random which object to give a task
+        foreach(KeyValuePair<string, GameObject[]> pair in allInteractableObjects)
+        {
+            List<GameObject> allObjects = pair.Value.ToList();
+            List<GameObject> allObjectsAdded = new List<GameObject>();
+
+            int amount = 0;
+            switch(pair.Key)
+            {
+                case "StreetLamp":
+                    amount = 15;
+                    break;
+                case "ManHole":
+                    amount = 8;
+                    break;
+                case "Tree":
+                    amount = 20;
+                    break;
+                case "SolarPanel":
+                    amount = 2;
+                    break;
+                default:
+                    amount = 10;
+                    break;
+            }
+
+            for (int i = 0; i < amount; i++)
+            {
+                int index = UnityEngine.Random.Range(0, allObjects.Count - 1);
+                GameObject specificGameObject = allObjects[index];
+                gameObjectsWithTasks.Add(specificGameObject);
+                allObjects.RemoveAt(index);
+                allObjectsAdded.Add(specificGameObject);
+            }
+
+            AddTasksToObject(allObjectsAdded, pair.Key);
+        }
+    }
+
+    public void MiniGameManager_OnGameOver(InteractableObject interactableObject)
+    {
+        //TODO: Arbitrary number
+        if (interactableObject.AmountTries >= 2)
+        {
+            interactableObject.gameObject.GetComponent<MeshRenderer>().material = failedMaterial;
+            Destroy(interactableObject);
+        }
+
+        interactableObject.AmountTries++;
+    }
+
+    public void MiniGameManager_OnGameWon(InteractableObject interactableObject)
+    {
+        //TODO: Change to better stuff
+        interactableObject.gameObject.GetComponent<MeshRenderer>().material = fixedMaterial;
+        Destroy(interactableObject);
+    }
+
+
+}
```

Comment:
I really like the general idea, but the problem that I have is that whenever I want to add a new mini game, I would have to change and add things in a lot of different places, and if everyone edits this file to add mini games, we might have a problem with conflicts sooner or later. (My point being is that this class should be as much as possible independent of the mini games, if I have 1 or 10 mini-games, it should act the same way without extra coding. (Also, this is dependent on the mini games right now, changing the mini-games might break this class which is a big negative. (See the notes for InteractableObject.cs for more)

Reply:
Yeah I agree. I made it as scalable as I could. There are some manual stuff to do when adding a mini-game, but now it's better. Lemme know if I shoudl try to make it more scalable (although I don't think I wouldn know how)